sycl/doc/extensions/ExplicitSIMD/dpcpp-explicit-simd.md

```diff
@@ -0,0 +1,776 @@
+  // m1 is a reference to the location of v1
+  // interpreted as a matrix 4x4 of ints.
+  // ...
+  auto m2 = v1.format<char, 4, 16>( );
+  // m2 is a reference to the location of v1
+  // interpreted as a matrix 4x16 of chars.
+```
+
+To model predicated move, Explicit SIMD provides the following merge functions:
+- ```merge(value_type Val, mask_type Mask)```: this merge operation takes one source operand **Val** and a mask **Mask** defined as unsigned short vector of the same length. The semantic is that if the LSB of an element of **Mask** is set, then the corresponding data element of **Val** is copied to the corresponding position in the calling simd object.
+
+```cpp
+  simd<int, 4>   m, src;
+  simd<unsigned short, 4> mask;
+  // ...
+  m.merge(src, mask);
+  // m           src         mask      --->  m
+  // 2 2 2 2     4 4 4 4     1 1 0 1         4 4 2 4
+```
+
+- ```merge(value_type Val1, value_type Val2, mask_type Mask)```: this merge operation takes two source operands **Val1** and **Val2** as well as a simd mask. The semantic is that if the LSB of an element of **Mask** is set, then the corresponding data element of **Val1** is copied to the corresponding position in the calling simd object. Otherwise the corresponding data element of **Val2** is copied to the corresponding position in the calling simd object.
+
+```cpp
+  simd<int,4>   m, src1, src2;
+  simd<unsigned short, 4> mask;
+  // ...
+  m.merge(src1, src2, mask);
+  // m           src1        src2         mask        --->  m
+  // 2 2 2 2     4 4 4 4     3 3 3 3      1 1 0 1           4 4 3 4
+```
+### `simd_view` class
+
+The ```sycl::intel::gpu::simd_view``` represents a "window" into existing simd object,
+through which a part of the original object can be read or modified. This is a
+syntactic convenience feature to reduce verbosity when accessing sub-regions of
+simd objects. **RegionTy** describes the window shape and can be 1D or 2D,
+**BaseTy** is the original simd object type, which can be a ```simd_view```
+itself.
+
+```simd_view``` allows to model hierarchical "views" of the parent ```simd```
+object's parts, read/modify its elements through the views. Views can be of
+different shapes and dimensions as illustrated below (`auto` resolves to a
+`simd_view` instantiation):
+
+<p align="center">
+<img src="images/simd_view.svg" title="1D select example" width="800" height="300"/>
+</p>
+
+```cpp
+namespace sycl {
+namespace intel {
+namespace gpu {
+template <typename BaseTy, typename RegionTy> class simd_view {
+public:
+  using ShapeTy = typename shape_type<RegionTy>::type;
+  static constexpr int length = ShapeTy::Size_x * ShapeTy::Size_y;
+  using value_type = simd<typename ShapeTy::element_type, length>;
+  using vector_type = vector_type_t<typename ShapeTy::element_type, length>;
+  using region_type = RegionTy;
+  using element_type = typename ShapeTy::element_type;
+
+  // Constructors.
+  simd_view(BaseTy &Base, RegionTy Region);
```

Comment:
Why is this and the next constructor public? Those should only be called internally by e.g. `select`, correct?

Reply:
added TODO to the headers